### ISSUE TITLE
Enhance Responsive Design and Improve Character Validation

### ISSUE THREAD

**Issue by S0KPiseth** [S0KPiseth/AlphaKeys#1]
### Description:
We need to enhance the responsive design for mobile devices and address the issue with incorrect character input handling.

1.  **Responsive Design:**

      - The mobile version of the site needs better optimization. Some elements are not properly aligned or scaled on smaller screens. The layout should be refined to ensure a seamless experience across all device sizes.
2. **Incorrect Character Input Handling:**

     - Currently, when a user types an incorrect character, it’s deleted, but sometimes the wrong character is allowed to remain in the input field instead of being properly deleted. This behavior needs to be fixed to ensure the correct character is always removed and replaced when necessary.
### Expected Behavior:

- The mobile version should display properly with improved alignment and scaling.
- When an incorrect character is typed, it should be deleted and replaced with the correct one without any inconsistencies.
### Steps to Reproduce:

- Type a practice word on both desktop and mobile versions.
- Type an incorrect character.
- Observe that, occasionally, the wrong character is not deleted correctly in the input field.
- Check the mobile layout for misalignment or scaling issues.

**Comment by AriajSarkar** [comment]
Before I start making code changes, I have a quick question: Can I use Tailwind CSS? It's excellent for creating responsive designs.
Additionally, I would like to improve the code structure by breaking it into smaller parts and moving reusable logic into utility files.

I also suggest that we consider using TypeScript. While JavaScript is certainly usable, TypeScript provides type safety, which can be very beneficial.

If you want...

**Comment by S0KPiseth** [comment]
Thanks for your interest in the project; I really appreciate it. I agree that using Tailwind CSS and TypeScript would be a great choice for the project. However, since I’m still new to React and haven’t learned Tailwind or TypeScript yet, I’m a bit concerned about managing the code effectively. Would it be alright if we stick with JavaScript and Bootstrap for responsive design for now?
And for improving the code structure, I’m completely fine with that approach.

**Comment by AriajSarkar** [comment]
Yes, sure, I am okay with it. I just thought you knew all the stuff and were still approaching the old method, which is why I mentioned it. I'll start the changes from tomorrow. 👌, Welcome.
